core/src/avm2/globals/flash/geom/Matrix.as

```diff
@@ -16,6 +19,103 @@ package flash.geom {
 			this.ty = ty;
 		}
 
+		public function clone():Matrix {
+			return new Matrix(this.a,this.b,this.c,this.d,this.tx,this.ty);
+		}
+
+		public function concat(m:Matrix):void {
+			var a = m.a * this.a + m.c * this.b;
+			var b = m.b * this.a + m.d * this.b;
+			var c = m.a * this.c + m.c * this.d;
+			var d = m.b * this.c + m.d * this.d;
+			var tx = m.a * this.tx + m.c * this.ty + m.tx;
+			var ty = m.b * this.tx + m.d * this.ty + m.ty;
+
+			this.a = a;
+			this.b = b;
+			this.c = c;
+			this.d = d;
+			this.tx = tx;
+			this.ty = ty;
+		}
+
+		public function copyColumnFrom(column:uint, vector3D:Vector3D):void {
+			// For some reason these methods are identical
```

Comment:
I haven't had time yet to verify this, or to see if you've added sufficient tests for this (I will soon, hopefully), but are you 100% extra sure about this, being a FP bug?

Reply:
There is tests for this method in Test.as (lines 389-410). Yes, I tested it multiple times during development in FP 11.2, and retested it after in FP 32 with the same results.

Reply:
I see, okay!
Weird how even in OpenFL there is some asymmetric accidental identity: https://github.com/openfl/openfl-native/commit/cf42d88e05c62bb26fdb6ed269f0761e4bf622d4

But only in the `From` methods, not in the `To` methods...

Reply:
Still, maybe a more noticeable `FP BUG` marker comment could be added. Not sure if we have any existing convention for this already.

Reply:
Added `FP BUG` note